### PR TITLE
Remove NeoForgeSPI dependency

### PR DESCRIPTION
**Pull request by Technici4n**
The indirection complicates matters for no benefit.

**Requires a major version bump.**

See https://github.com/neoforged/FancyModLoader/pull/106 for the corresponding FML PR.

### CODE REVIEW

File: src/main/java/net/neoforged/coremod/ICoreModFile.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+package net.neoforged.coremod;
+
+import java.io.*;
```
Comment (shartte) [inline src/main/java/net/neoforged/coremod/ICoreModFile.java]: Wildcard import bad.

**Comment by shartte** [comment]
General thoughts: we should use this opportunity to make it clear this is the Coremod *Scripting* Engine.
As such: `ICoreModFile` -> `ICoreModScriptSource` 
`CoreModEngine` -> `CoreModScriptingEngine`

Thoughts?

**Comment by Technici4n** [comment]
> General thoughts: we should use this opportunity to make it clear this is the Coremod _Scripting_ Engine. As such: `ICoreModFile` -> `ICoreModScriptSource` `CoreModEngine` -> `CoreModScriptingEngine`
> 
> Thoughts?

why not - for a separate PR though. ;)

**Comment by shartte** [comment]
> > General thoughts: we should use this opportunity to make it clear this is the Coremod _Scripting_ Engine. As such: `ICoreModFile` -> `ICoreModScriptSource` `CoreModEngine` -> `CoreModScriptingEngine`
> > Thoughts?
> 
> why not - for a separate PR though. ;)

Thought about this some more: If this PR leads to a breaking change release, shouldn't we actually include *all* breaking changes since any subsequent breaks would also require a new breaking release? :-P

**Review by shartte — APPROVED** [review]

**Pull request merged**